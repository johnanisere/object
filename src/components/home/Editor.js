import React from "react"
import PropTypes from "prop-types"
import ImageSlider from "./ImageSlider"
import Container from "../../blocks/FlexibleContainer";


class App extends React.Component {
    render() {
        const { handleMouseMove, handleClick, displayData, cursor, element, data, onSwitchImage, active, displayImage } = this.props;
        return (
            <Container background={true} height="100%" flex={3} border={true} direction="column">
                <Container header={true} />
                <Container editor={true} cursor={cursor} >

                    <figure className="img-holder" onClick={handleClick} onMouseMove={handleMouseMove} >
                        <img src={displayImage} alt="The Pulpit Rock" />

                        {
                            displayData.map(({ Figure }, key) => <Figure key={key} ></Figure>)
                        }
                        {element && element}
                    </figure>
                </Container>
                <ImageSlider data={data} onSwitchImage={onSwitchImage} active={active} />
            </Container>
        );
    }
}


App.propTypes = {
    onSwitchImage: PropTypes.func.isRequired,
    handleMouseMove: PropTypes.func.isRequired,
    displayImage: PropTypes.string.isRequired,
    handleClick: PropTypes.func.isRequired,
    displayData: PropTypes.array.isRequired,
    cursor: PropTypes.string.isRequired,
    active: PropTypes.number.isRequired,
    data: PropTypes.array.isRequired,
    element: PropTypes.node,
}
export default App