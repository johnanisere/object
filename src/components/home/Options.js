import React from "react"
import PropTypes from "prop-types"
import Container from "../../blocks/FlexibleContainer"
import Objects from "./Objects"
import Classes from "./Classes"
import Tags from "./Tags"

const App = ({ displayData }) => {
    return (
        <Container background={true} height="100%" flex={1} direction="column">
            <Objects displayData={displayData} />
            <Classes />
            <Tags />
        </Container>
    )
}

App.propTypes = {
    displayData: PropTypes.array.isRequired,
}
export default App