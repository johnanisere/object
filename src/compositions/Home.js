import React from "react";

import Home from "../components/home"
import Layout from "../components/Layout";

class App extends React.Component {
  render() {
    return (
      <Layout>
        <Home />
      </Layout>
    );
  }
}

export default App;
