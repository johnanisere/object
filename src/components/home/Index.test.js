
import React from "react";
import { shallow } from "enzyme";

import Home from "./index"
import Editor from "./Editor"
import Options from "./Options"

it("Home component exists", () => {
    const wrapper = shallow(<Home />);
    expect(wrapper.exists()).toBe(true);
});

it("Home component renders", () => {
    const wrapper = shallow(<Home />);
    expect(wrapper).toMatchSnapshot();
});

it(`Home should contain "Options", u"Editor", `, () => {
    const wrapper = shallow(<Home />);

    expect(wrapper.find(Editor)).toHaveLength(1);
    expect(wrapper.find(Options)).toHaveLength(1);
});