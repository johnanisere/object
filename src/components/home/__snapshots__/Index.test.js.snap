// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders 1`] = `ShallowWrapper {}`;
